libs/native/src/main/java/org/elasticsearch/nativeaccess/AbstractNativeAccess.java

```diff
@@ -10,18 +10,36 @@
 
 import org.elasticsearch.logging.LogManager;
 import org.elasticsearch.logging.Logger;
+import org.elasticsearch.nativeaccess.lib.JavaLibrary;
+import org.elasticsearch.nativeaccess.lib.NativeLibraryProvider;
+import org.elasticsearch.nativeaccess.lib.ZstdLibrary;
 
 abstract class AbstractNativeAccess implements NativeAccess {
 
     protected static final Logger logger = LogManager.getLogger(NativeAccess.class);
 
     private final String name;
+    private final JavaLibrary javaLib;
+    private final Zstd zstd;
 
-    protected AbstractNativeAccess(String name) {
+    protected AbstractNativeAccess(String name, NativeLibraryProvider libraryProvider) {
         this.name = name;
+        this.javaLib = libraryProvider.getLibrary(JavaLibrary.class);
```

Comment:
I'm a bit confused by where things are placed in the hierarchy; 
I think the rationale is that we implement as much as possible of the `NativeAccess` interface here, delegating to the library provider which will use the correct Java native "bridge" (JNA or JDK). 
And we leave out (implement in derived classes) only those functions that differ across OS/library? (e.g. are implemented by different `NativeLibrary`es?).
Is that correct or am I missing something?

Reply:
Your summary is correct.

This "JavaLibrary" is a special case that I didn't have a better terminology for. It's not actually a library, but it's something that we need to differ between the two bridge implementations, namely how do you get a native buffer that can be closed.

Reply:
Yeah, that naming is a bit weird. Would something like `ByteBufferLibrary` make more sense?

Reply:
I named it using "Java" because there will likely be other things we want to differentiate, it's not just for ByteBuffer. I wanted a future place for any java related code with different impls for jna vs jdk.

Reply:
I wonder if `StandardLibrary` might be a good alternative. Consider this the end of my bike shedding, I'm also good with this as-is 😉

Reply:
I'm going to keep it JavaLibrary for now, but will consider renaming in the future if/as we add more to it.